macros/anaUltraThreshold.py

```diff
@@ -53,121 +64,215 @@
+    vSum[vfat].Write()
     pass
 canv.SaveAs(filename+'/ThreshSummary.png')
 
+canv_proj = r.TCanvas('canv_proj', 'canv_proj', 500*8, 500*3)
+canv_proj.Divide(8,3)
+r.gStyle.SetOptStat(0)
+print 'Saving File'
+for vfat in range(0,24):
+    r.gStyle.SetOptStat(0)
+    canv_proj.cd(vfat+1)
+    r.gPad.SetLogy()
+    vSum[vfat].ProjectionY().Draw()
+    pass
+canv_proj.SaveAs(filename+'/VFATSummary.png')
+
 #Subtracting off the hot channels, so the projection shows only usable ones.
-for i in range(0,24):
-    for j in range(0,128):
-       if hot_channels[i][j] is True:
-           print 'VFAT %i Strip %i is noisy'%(i,j)
-           for binY in range(VT1_MAX+1):
-               bin = vSum[i].GetBin(j+1, binY)
-               content = vSum[i].GetBinContent(bin)
-               vSum[i].SetBinContent(bin, 0)
-               pass
-           pass
-       pass
+print "Subtracting off hot channels"
+for vfat in range(0,24):
+    if (vfat_mask >> vfat) & 0x1: continue
+    for chan in range(0,vSum[vfat].GetNbinsX()):
+        isHotChan = hot_channels[vfat][chan]
+
+        if options.chConfigKnown:
+	          isHotChan = (isHotChan or dict_vfatTrimMaskData[vfat][chan]['mask'])
+	          pass
+
+        if isHotChan:
+            print 'VFAT %i Strip %i is noisy'%(vfat,chan)
+            for thresh in range(VT1_MAX+1):
+                vSum[vfat].SetBinContent(chan, thresh, 0)
+                pass
+            pass
+        pass
     pass
 
+#Save output with new hot channels subtracted off
 canv_pruned = r.TCanvas('canv_pruned','canv_pruned',500*8,500*3)
 canv_pruned.Divide(8,3)
 r.gStyle.SetOptStat(0)
 print 'Saving File'
-for i in range(0,24):
+for vfat in range(0,24):
     r.gStyle.SetOptStat(0)
-    canv_pruned.cd(i+1)
-    vSum[i].Draw('colz')
-    vSum[i].Write()
+    canv_pruned.cd(vfat+1)
+    vSum[vfat].Draw('colz')
+    vSum[vfat].Write()
     pass
 canv_pruned.SaveAs(filename+'/ThreshPrunedSummary.png')
 
-
-canv_proj = r.TCanvas('canv_proj', 'canv_proj', 500*8, 500*3)
+canv_proj = r.TCanvas('canv_proj_pruned', 'canv_proj_pruned', 500*8, 500*3)
 canv_proj.Divide(8,3)
 r.gStyle.SetOptStat(0)
 print 'Saving File'
-for i in range(0,24):
+for vfat in range(0,24):
     r.gStyle.SetOptStat(0)
-    canv_proj.cd(i+1)
+    canv_proj.cd(vfat+1)
     r.gPad.SetLogy()
-    vSum[i].ProjectionY().Draw()
-    vSum[i].ProjectionY().Write()
+    vSum[vfat].ProjectionY().Draw()
+    vSum[vfat].ProjectionY().Write()
     pass
-canv_proj.SaveAs(filename+'/VFATSummary.png')
+canv_proj.SaveAs(filename+'/VFATPrunedSummary.png')
 #outF.Close()
 
 #Now determine what VT1 to use for configuration.  The first threshold bin with no entries for now.
 #Make a text file readable by TTree::ReadFile
-vt1 = []
-vt1new = []
-for i in range(0,24):
-    for j in range(VT1_MAX+1,0,-1):
-        #original method
-        if vSum[i].ProjectionY().GetBinContent(j+1) > 10.0:
-            print 'vt1 for VFAT %i found'%i
-            vt1.append(j+1)
+vt1 = {}
```

Comment:
* ```vt1new``` is obviated by the change that @bdorney is proposing
* be that as it may, one needs to ensure that one's code base is up to date before doing pull requests
  * standard operating procedure:
   * PR from central github to your fork
   * fetch from your fork to your working area
   * merge upstream changes into your feature branch